### PR TITLE
Pip installable

### PR DESCRIPTION
**Pull request by parenthetical-e**
This PR has two parts.
1. I added the files needed to make the `wavetorch` core pip installable (i.e, `setup.py` and `requirements.txt`.
2. I made some minimal fixes so `Cell` plays well with pytorch 1.2 (i.e. a type case, and removal an an inplace op in `step()`). I'd be _very_ curious and grateful if you fine folks would confirm these changes are safe and ok, even if you don't want this PR. I'm using them for new work...

_Note_: I commented out the old/original code for the changes in point 2, really just for my reference. I'd be happy to clean that cruft up if this PR moves forward.

### CODE REVIEW

**Comment by parenthetical-e** [comment]
Right. Also, for lack of better option my email and name or in the `setup.py` file. I'd expect you'd change that to someone there. Not trying to claim credit. :)

**Comment by ianwilliamson** [comment]
Thanks for this! I've been pretty busy but I'll take a look at this in more detail as soon as possible.

File: wavetorch/core/cell.py [hunk 1]
```diff
@@ -203,14 +207,22 @@ def step(self, x, y1, y2, c_linear, proj_rho):
...
         # Insert the source
-        y[:, self.src_x, self.src_y] = y[:, self.src_x, self.src_y] + x.expand_as(y[:, self.src_x, self.src_y])
+        # Orginal code (v0.1)
+        # y[:, self.src_x, self.src_y] = y[:, self.src_x, self.src_y] + x.expand_as(y[:, self.src_x, self.src_y])
         
+        # Modified for pytorch > 1 (inplace bug in orginal)
+        x_in = torch.zeros(y.size()).detach()
+        x_in[:, self.src_x, self.src_y] = x
+        
+        y = y + x_in
```
Comment (ianwilliamson) [inline wavetorch/core/cell.py]: Can you clarify here what the issue was with the original code? Is this something that should work on pytorch 1.0 but not on 1.2? I don't seem to be getting any explicit errors on either. Or is the issue that the gradients are just wrong?

Comment (ianwilliamson) [reply wavetorch/core/cell.py]: Nevermind, I was able to reproduce the issue.

Comment (ianwilliamson) [reply wavetorch/core/cell.py]: This appears to be an issue only when using a line source. I'm guessing you encountered it when running `propagate.py` but probably not when running the vowel training?

File: wavetorch/core/cell.py [hunk 2]
```diff
@@ -121,7 +123,8 @@ def __init__(self, dt : float, Nx : int, Ny : int,
         if design_region is not None:
             # Use specified design region
             assert design_region.shape == (Nx, Ny), "Design region mask dims must match spatial dims"
-            self.register_buffer("design_region", design_region * (self.b_boundary == 0))
+            # self.register_buffer("design_region", design_region * (self.b_boundary == 0))
+            self.register_buffer("design_region", design_region * (self.b_boundary == 0).type(torch.uint8))
```
Comment (ianwilliamson) [inline wavetorch/core/cell.py]: This is probably best practice for a mask, but I just want to make sure... were you actually encountering an error with the old code?

File: wavetorch/core/cell.py (hunk 1 above)
Comment (parenthetical-e) [reply wavetorch/core/cell.py]: Exactly.

File: wavetorch/core/cell.py (hunk 2 above)
Comment (parenthetical-e) [reply wavetorch/core/cell.py]: I was. A type error.

**Comment by ianwilliamson** [comment]
So I've pushed/tagged a new version of this package. The changes you suggested in this pull request were easier to just incorporate myself given the amount of refactoring I did. Therefore, I'm going to close this pull request. I gave you a shout out in the code for the fix to the source term in pytorch 1.2. Thanks again for that!
